web/pdf_print_service.js

```diff
@@ -27,8 +29,28 @@ function renderPage(
   pageNumber,
   size,
   printResolution,
-  optionalContentConfigPromise
+  optionalContentConfigPromise,
+  printContainer
 ) {
+  if (pdfDocument.isPureXfa) {
+    const page = document.createElement("div");
+    page.setAttribute("class", "xfaPrintedPage");
+    printContainer.appendChild(page);
+
+    return pdfDocument.getPage(pageNumber).then(pdfPage => {
+      const builder = new DefaultXfaLayerFactory().createXfaLayerBuilder(
```

Comment:
I know that this is still a WIP, but for future reference:
We should probably initialize/re-use just *one* instance of `DefaultXfaLayerFactory` for all pages.